core/src/main/java/feast/core/service/JobCoordinatorService.java

```diff
@@ -195,8 +195,13 @@ private void updateFeatureSetStatuses(List<JobUpdateTask> jobUpdateTasks) {
     return Optional.of(jobs.get(0));
   }
 
-  // TODO: Put in a util somewhere?
-  private static List<FeatureSet> featureSetsFromProto(List<FeatureSetProto.FeatureSet> protos) {
-    return protos.stream().map(FeatureSet::fromProto).collect(Collectors.toList());
+  private List<FeatureSet> featureSetsFromProto(List<FeatureSetProto.FeatureSet> protos) {
+    return protos.stream()
+        .map(FeatureSetProto.FeatureSet::getSpec)
+        .map(
+            fs ->
+                featureSetRepository.findFeatureSetByNameAndProject_NameAndVersion(
+                    fs.getName(), fs.getProject(), fs.getVersion()))
```

Comment:
Could we do this without N+1 queries?

Reply:
It's a little hard unless i refactor the spec service so that the `listFeatureSets` logic is available to the job service :/ Any suggestions here?

Reply:
I'm not that keen on creating a composite id for featuresets since many other tables join on featuresets, and join tables really add to the complexity of the data model.

Reply:
> I'm not that keen on creating a composite id for featuresets since many other tables join on featuresets, and join tables really add to the complexity of the data model.

Was this comment meant to be somewhere else?

Reply:
> It's a little hard unless i refactor the spec service so that the `listFeatureSets` logic is available to the job service :/ Any suggestions here?

Maybe let's just stick an `OPTIMIZE:` comment on it, and add it to the scope of #664.

Reply:
ok cool

Reply:
> It's a little hard unless i refactor the spec service so that the listFeatureSets logic is available to the job service :/ Any suggestions here?

I think we should be able to create another query right? You already have the `featureSetRepository`,